### PR TITLE
Configure isort to use the Black profile, recursively isort the `thinc` module

### PR DESCRIPTION
**Pull request by danieldk**
## Description

Configure isort to use the Black profile, recursively isort the `thinc` module. Also fix a cyclic import.

### Types of change

Maintenance

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -9,3 +9,6 @@ requires = [
     "numpy>=1.15.0",
 ]
 build-backend = "setuptools.build_meta"
+
+[tool.isort]
+profile = "black"
```
Comment (rmitsch) [inline pyproject.toml:14]: So if I understand it correctly, this is one of three manual changes, right?

File: requirements.txt [hunk 2]
```diff
@@ -36,3 +36,4 @@ nbformat>=5.0.4,<5.2.0
 # Test to_disk/from_disk against pathlib.Path subclasses
 pathy>=0.3.5
 black>=22.0,<23.0
+isort>=5.0,<6.0
```
Comment (rmitsch) [inline requirements.txt:39]: This being the second manual change.

**Review by rmitsch — COMMENTED** [review]
Can you point out where else you changed things manually?

**Comment by danieldk** [comment]
These are the manual changes besides adding the `isort` configuration to `pyproject.toml`:

https://github.com/explosion/thinc/pull/880/files/6b03cbd225b508f14e31f197cd62575d379e474b..29cace2441ac194be2bb80c1a0539455aea7b271

**Review by rmitsch — APPROVED** [review]
Same as for the spaCy PR - manual changes look good.

**Pull request merged**